src/classes/BGE_BatchGiftEntry_UTIL.cls

```diff
@@ -36,6 +36,82 @@
+    * @return List<String> list of DataImport__c field api names
+    */
+    public static List<String> getDataImportFields(Id batchId) {
+        return getDataImportFields(batchId, true);
+    }
+
+    /*******************************************************************************************************
+    * @description returns a list of DataImport__c fields the Batch Gift Entry UI needs in SOQL
+    * @param batchId the batch for which to get all fields for soql
+    * @param includeRelationshipFields whether to include relationship fields in the returned list
+    * @return List<String> list of DataImport__c field api names
+    */
+    public static List<String> getDataImportFields(Id batchId, Boolean includeRelationshipFields) {
+
+        List<String> fields = getDataImportFields(includeRelationshipFields);
+        fields.addAll(getActiveFieldNamesFromBatch(batchId));
+
+        return fields;
+    }
+
+    /*******************************************************************************************************
+    * @description returns the subset of DataImport__c fields that are part of every batch
+    * @param includeRelationshipFields whether to include relationship fields in the returned list
+    * @return List<String> list of DataImport__c field api names
+    */
+    private static List<String> getDataImportFields(Boolean includeRelationshipFields) {
+        List<String> fields = new List<String> {
+                'Id',
+                'Account1Imported__c',
+                'Contact1Imported__c',
+                'Donation_Donor__c',
+                'DonationImported__c',
+                'DonationImportStatus__c',
+                'FailureInformation__c',
+                'NPSP_Data_Import_Batch__c',
+                'PaymentImported__c',
+                'PaymentImportStatus__c',
+                'Status__c'
+        };
+
+        if (includeRelationshipFields) {
+            fields.addAll(new List<String>{
+                'Account1Imported__r.Name',
+                'Contact1Imported__r.Name',
+                'DonationImported__r.Name',
+                'PaymentImported__r.Name'
+            });
+        }
+
+        return fields;
+    }
+
+    /*******************************************************************************************************
+    * @description
+    * @param batchId: ID of the NPSP_Data_Import_Batch__c
+    * @return list of field api names
+    */
+    public static List<String> getActiveFieldNamesFromBatch(Id batchId) {
+
+        DataImportBatch__c batch = [SELECT Active_Fields__c FROM DataImportBatch__c WHERE Id = :batchId];
+        String activeFieldsJSON = batch.Active_Fields__c;
+        List<String> activeFieldNames = new List<String>();
+
+        if (activeFieldsJSON != null) {
+            List<BGE_ConfigurationWizard_CTRL.BGEField> activeFields = (List<BGE_ConfigurationWizard_CTRL.BGEField>)JSON.deserialize(activeFieldsJSON, List<BGE_ConfigurationWizard_CTRL.BGEField>.class);
```

Comment:
Deserializing in Apex like this makes me think we should reconsider child objects...